src/porepy/params/data.py

```diff
@@ -0,0 +1,510 @@
+        s += str(list(self.__dict__.keys()))
+        return s
+
+    def _get_physics(self, obj):
+        if isinstance(obj, Solver):
+            if not hasattr(obj, 'physics'):
+                raise AttributeError('Solver object should have attribute physics')
+            s = obj.physics.strip().lower()
+        elif isinstance(obj, str):
+            s = obj.strip().lower()
+        else:
+            raise ValueError('Expected str or Solver object')
+
+        if not s in self.known_physics:
+            # Give a warning if the physics keyword is unknown
+            warnings.warn('Unknown physics ' + s)
+        return s
+
+#------------- Start of definition of parameters -------------------------
+
+#------------- Constants
+
+#--------------- Biot Alpha ---------------------------------------
+    def _get_biot_alpha(self):
+        if hasattr(self, '_biot_alpha'):
+            return self._biot_alpha
+        else:
+            return 1
+    def _set_biot_alpha(self, val):
+        if val < 0 or val > 1:
+            raise ValueError('Biot\'s constant should be between 0 and 1')
+        self._biot_alpha = val
+    biot_alpha = property(_get_biot_alpha, _set_biot_alpha)
+
+#--------------- Fluid viscosity ------------------------------------
+    def _get_fluid_viscosity(self):
+        if hasattr(self, '_fluid_viscosity'):
+            return self._fluid_viscosity
+        else:
+            return 1
+
+    def _set_fluid_viscosity(self, val):
+        if val <= 0:
+            raise ValueError('Fluid viscosity should be positive')
+        self._fluid_viscosity = val
+    fluid_viscosity = property(_get_fluid_viscosity, _set_fluid_viscosity)
+
+#----------- Fluid compressibility
+    def _get_fluid_compr(self):
+        if hasattr(self, '_fluid_compr'):
+            return self._fluid_compr
+        else:
+            return 0.
+
+    def _set_fluid_compr(self, val):
+        if val < 0:
+            raise ValueError('Fluid compressibility should be non-negative')
+        self._fluid_compr = val
+    fluid_compr = property(_get_fluid_compr, _set_fluid_compr)
+
+#-------------------- Cell-wise quantities below here --------------------
+
+#------------------ Aperture -----------------
+
+    def _get_aperture(self, default=1):
+        """ double or array_like
+        Cell-wise quantity representing fracture aperture (really, height of
+        surpressed dimensions). Set as either a np.ndarray, or a scalar
+        (uniform) value. Always returned as np.ndarray.
+        """
+        if not hasattr(self, '_apertures'):
+            return default * np.ones(self._num_cells)
+
+        if isinstance(self._aperture, np.ndarray):
+            # Hope that the user did not initialize as array with wrong size
+            return self._apertures
+        else:
+            return self._apertures * np.ones(self._num_cells)
+
+    def _set_aperture(self, val):
+        if (isinstance(val, np.ndarray) and np.any(val < 0)) or val < 0:
```

Comment:
This does not work if you send inn an array with only positive values.

What is the reason for not doing: `if np.any(val < 0):` ? This seems to work just fine for me